### PR TITLE
fix(overlay): adjust max heights to always fit viewport

### LINKED ISSUE #2882: Cloud deploy modal is cut off from the window.
__Describe the Bug__

Choose to deploy a BPMN diagram via cloud we could see that the deploy diagram modal is cut-off from the windows.

__Steps to Reproduce__

1. Create/Open a BPMN diagram in the Camunda 8 Platform
2. Choose to deploy in any of the cloud versions and choose Camunda Platform self-managed and oAuth for authentication.
3. We could see that the deploy diagram modal is cut off from the windows.

__Expected Behavior__

The depoly diagram modal should fit inside the window.

__Environment__

- OS: [ Windows 10]
- Camunda Modeler Version: [5.0.0 Nightly]
- Execution Platform: [Camunda Cloud]
323562a893633abd34adab495812bf0a

### PR DESCRIPTION
**Pull request by smbea**
Closes #2882

https://user-images.githubusercontent.com/25825387/162963040-9987c8d4-82ac-45ec-937d-bf34cfb780a6.mov

### CODE REVIEW

File: client/src/plugins/report-feedback/ReportFeedbackOverlay.js [hunk 1]
```diff
@@ -27,7 +27,6 @@ export function ReportFeedbackOverlay(props) {
     <Overlay
       anchor={ props.anchor }
       onClose={ props.onClose }
-      maxHeight="calc(100vh - 250px)"
```
Comment (smbea) [inline client/src/plugins/report-feedback/ReportFeedbackOverlay.js:30]: **Explanation:** this was causing the the report feedback overlay to shorten and have scroll, even though there was plenty of space still. The height of this overlay is a lot smaller than `MIN_HEIGHT` of the Modeler, meaning we don't need to set this.

**Comment by andreasgeier** [comment]
Expected:

https://user-images.githubusercontent.com/938244/162776202-0c636b66-9d1b-4b7d-b96a-69f9b05cbb19.mov

**Comment by smbea** [comment]
@andreasgeier I updated the screen recording accordingly

**Review by andreasgeier — APPROVED** [review]
Works as expected 👍

File: client/src/shared/ui/overlay/Overlay.less [hunk 2]
```diff
@@ -84,8 +84,10 @@
   
   fieldset {
     border: none;
-    margin: 0 0 5px 0;
-    padding: 0;
+    margin: 0 -12px 5px 0;
+    padding: 0 12px 0 0;
```
Comment (pinussilvestrus) [inline client/src/shared/ui/overlay/Overlay.less:88]: Why do we need the negative margin here (adding the padding again)? Is it a CSS hack?

File: client/src/shared/ui/overlay/Overlay.less [hunk 3]
```diff
@@ -84,8 +84,10 @@
...
+    max-height: calc(100vh - 176px);
+    overflow-y: scroll;
```
Comment (pinussilvestrus) [inline client/src/shared/ui/overlay/Overlay.less:90]: On my machine, the scrollbar is visible even though it isn't necessary

image

I think that can be fixed via `overflow-y: auto`.

```suggestion
    overflow-y: auto;
```

**Review by pinussilvestrus — CHANGES_REQUESTED** [review]
Looks good so far 👍

File: client/src/shared/ui/overlay/Overlay.less (hunk 3 above)
Comment (smbea) [reply client/src/shared/ui/overlay/Overlay.less:90]: Ah that doesn't happen on mine. But I will switch to this.

File: client/src/shared/ui/overlay/Overlay.less (hunk 2 above)
Comment (smbea) [reply client/src/shared/ui/overlay/Overlay.less:88]: I hope my explanation was clear enough to understand

Comment (pinussilvestrus) [reply client/src/shared/ui/overlay/Overlay.less:88]: Got it, thanks :)

**Review by pinussilvestrus — APPROVED** [review]
LGTM 👍

**Pull request merged**